### PR TITLE
Migrate Auth in Spring SDKs

### PR DESCRIPTION
**Pull request by rujche**
# Description

Migrate Auth in Spring SDKs

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-integration-tests/src/test/java/com/azure/spring/cloud/integration/tests/keyvault/secret/KeyVaultSecretIT.java [hunk 1]
```diff
@@ -24,10 +27,26 @@ public class KeyVaultSecretIT {
...
         client.setSecret(NAME, VALUE);
         KeyVaultSecret secret = client.getSecret(NAME);
         Assertions.assertEquals(VALUE, secret.getValue());
-        LOGGER.info("KeyVaultSecretIT end.");
+        LOGGER.info("testKeyVaultSecretOperation end.");
+    }
+
+    @Test
+    public void testKeyVaultSecretByDefaultAzureCredential() {
+        DefaultAzureCredential credential = new DefaultAzureCredentialBuilder()
```
Comment (g2vinay) [inline sdk/spring/spring-cloud-azure-integration-tests/src/test/java/com/azure/spring/cloud/integration/tests/keyvault/secret/KeyVaultSecretIT.java:39]: So, the credential logic to use FederatedAuth is as follows:

```java
public class TestUtil {

    /**
     * Gets a token credential for use in tests.
     * @param interceptorManager the interceptor manager
     * @return the TokenCredential
     */
    public static TokenCredential getTestTokenCredential(InterceptorManager interceptorManager) {
        if (interceptorManager.isLiveMode()) {
            return new AzurePowerShellCredentialBuilder().build();
        } else if (interceptorManager.isRecordMode()) {
            return new DefaultAzureCredentialBuilder().build();
        } else {
            return getPlaybackTokenCredential();
        }
    }

    /**
     * Gets a playback token credential for use in tests.
     * @return the TokenCredential
     */
    public static TokenCredential getPlaybackTokenCredential() {
        return new MockTokenCredential();
    }
}


```

File: sdk/spring/spring-cloud-azure-integration-tests/src/test/java/com/azure/spring/cloud/integration/tests/ApplicationConfiguration.java [hunk 2]
```diff
@@ -16,4 +21,9 @@
 @SpringBootConfiguration
 public class ApplicationConfiguration {
 
+    @Bean(name = DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME)
+    TokenCredential tokenCredential() {
+        return new AzurePowerShellCredentialBuilder().build();
```
Comment (rujche) [inline sdk/spring/spring-cloud-azure-integration-tests/src/test/java/com/azure/spring/cloud/integration/tests/ApplicationConfiguration.java:26]: FYI: https://github.com/Azure/azure-sdk-for-java/pull/40406/files#r1637109223

**Comment by rujche** [comment]
Hi, @saragluna , @g2vinay . Please help to review this PR.

**Review by saragluna — APPROVED** [review]

**Pull request merged**